### PR TITLE
Add core48 and remove 48way

### PR DESCRIPTION
**Pull request by opelpanfan**
Add core48 board config and remove 48way

### CODE REVIEW

File: firmware/config/boards/core48/knock_config.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#pragma once
+
+// Knock is on ADC3
+#define KNOCK_ADC ADCD3
+
+// knock 1 - pin PC0
+#define KNOCK_ADC_CH1 ADC_CHANNEL_IN10
+#define KNOCK_PIN_CH1 Gpio::C0
+
+// knock 2 - pin PC0
```
Comment (mck1117) [inline firmware/config/boards/core48/knock_config.h:14]: typo?

File: firmware/config/boards/core48/board_configuration.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	engineConfiguration->lps25BaroSensorSda = Gpio::B11;
+
+}
+
+
+void setBoardDefaultConfiguration(void) {
+	setInjectorPins();
+	setIgnitionPins();
+	setupDefaultSensorInputs();
+	setupVbatt();
```
Comment (mck1117) [inline firmware/config/boards/core48/board_configuration.cpp:175]: this should probably go in overrides, not default

File: firmware/config/boards/core48/board_configuration.cpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+
+	engineConfiguration->canWriteEnabled = true;
+	engineConfiguration->canReadEnabled = true;
+	engineConfiguration->canSleepPeriodMs = 50;
+
+	engineConfiguration->canBaudRate = B500KBPS;
+	engineConfiguration->can2BaudRate = B500KBPS;
+
+	engineConfiguration->sdCardCsPin = Gpio::B3;
```
Comment (mck1117) [inline firmware/config/boards/core48/board_configuration.cpp:185]: this should probably go in overrides, not default

File: firmware/config/boards/core48/board_configuration.cpp (hunk 2 above)
Comment (opelpanfan) [reply firmware/config/boards/core48/board_configuration.cpp]: moved vbat and sd card config to overwrites

File: firmware/config/boards/core48/board_configuration.cpp [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+static void setStepperConfig() {
+	engineConfiguration->idle.stepperDirectionPin = Gpio::A8;
+	engineConfiguration->idle.stepperStepPin = Gpio::C6;
+	engineConfiguration->stepperEnablePin = Gpio::C7;
+}
+
+
+static void setLedPins() {
+	// G12 is error LED, configured in board.mk
+	//engineConfiguration->communicationLedPin = Gpio::G13;
```
[the remark this answers is not in the document]
Comment (opelpanfan) [reply firmware/config/boards/core48/board_configuration.cpp]: noted and changed as per proteus board_configuration.cpp

File: firmware/config/boards/core48/knock_config.h (hunk 1 above)
Comment (opelpanfan) [reply firmware/config/boards/core48/knock_config.h]: Yes small copy/paste drama.

**Comment by mck1117** [comment]
Oh - one more thing - can you make the matching edits to `.github/workflows/build-firmware.yaml`? Let's see a green check for the board in the build gates!

File: firmware/config/boards/core48/board_configuration.cpp (hunk 3 above)
Comment (mck1117) [reply firmware/config/boards/core48/board_configuration.cpp:187]: ahh, got it

**Comment by opelpanfan** [comment]
Yes please if you could merge would be great.

**Comment by mck1117** [comment]
@opelpanfan once a build on master is done, go ahead and try out a bundle zip from the firmware action to see if it works!

**Comment by opelpanfan** [comment]
@mck1117 Thank you.

**Pull request merged**